zstd.go

```diff
@@ -67,11 +67,14 @@ func decompressSizeHint(src []byte) int {
 	}
 
 	hint := upperBound
-	if len(src) >= zstdFrameHeaderSizeMax {
+	if len(src) >= zstdFrameHeaderSizeMin {
```

Comment:
why not check the upper bound as well?

```go
len(src) >= zstdFrameHeaderSizeMin && len(src) <= zstdFrameHeaderSizeMax
```

Reply:
I'm not sure what you mean ? That check if just to make sure the header is in the payload but src will 99% of the time be bigger than `zstdFrameHeaderSizeMax` (and that's fine)

Reply:
👍 sorry I thought that `src` would only be the header, I didn't realize it was the full payload.